src/simulator/privacy_tracker.py

```diff
@@ -89,7 +112,26 @@ def privacy_consumption(self) -> PrivacyBudget:
         basic composition rule.  This will be expanded in the future
         to support advanced composition (see TODO #1 above).
         """
-        return PrivacyBudget(self._epsilon_sum, self._delta_sum)
+        # Take maximum privacy budget spent across all buckets.
+        max_epsilon = 0
+        max_delta = 0
+        for sampling_bucket_ in self._sampling_buckets_starting_points:
```

Comment:
logically we shall take maximum over all x in [0, 1), but it seems enough to look at just starting points? 

This is smart! Would you consider proving or explaining it here.

Reply:
Added a comment to clarify this.